x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -752,6 +755,7 @@ public void mayFinish() {
             if (expectedSlices.decrementAndGet() == 0) {
                 builder.setExecutables(executables);
                 builder.setStackFrames(stackFrames);
+                builder.totalFrames += totalInlineFrames.get();
```

Comment:
Can we please implement an `addTotalFrames` / `addInlineFrameCount` method and call that instead? In the stack trace handler we call the corresponding setter and we should not access the field directly here. Also, can you please add a comment why we add the inline frames as well (might help future readers :))?